service/handlers.go

```diff
@@ -0,0 +1,149 @@
+package service
+
+import (
+	"encoding/json"
+	"fmt"
+	log "github.com/Sirupsen/logrus"
+	"github.com/rancher/go-rancher/v2"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+type WebhookRequest map[string]interface{}
+
+func Construct(w http.ResponseWriter, r *http.Request) {
+	var webhookRequestData WebhookRequest
+	log.Infof("Construct Payload")
+	bytes, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Errorf("Construct failed with error: %v", err)
+		return
+	}
+	json.Unmarshal(bytes, &webhookRequestData)
+	_, err = ConstructPayload(webhookRequestData)
```

Comment:
You want the input map and output map to be completely separate.
Your goal should be this : if you pass in map X into Construct for driver Foo, you get out map Y that you can pass directly into Execute for driver Foo and it can successfully execute the webhook.